accounts/views.py

```diff
@@ -40,3 +50,109 @@ class UserLoginView(LoginView):
 
 class UserLogoutView(LogoutView):
     pass
+
+
+class UserProfileView(LoginRequiredMixin, DetailView):
+    template_name = "accounts/profile.html"
+    model = Profile
+
+    def get_queryset(self):
+        return super().get_queryset().select_related("user")
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = super().get_context_data(*args, **kwargs)
+        user = self.object.user
+        ctx["tweet_list"] = Tweet.objects.select_related("user").filter(user=user)
+        ctx["following_count"] = FriendShip.objects.filter(follower=user).count()
+        ctx["follower_count"] = FriendShip.objects.filter(following=user).count()
+        ctx["has_following_connection"] = (
+            FriendShip.objects.select_related("follower", "following")
+            .filter(follower=self.request.user, following=user)
+            .exists()
+        )
+
+        return ctx
+
+
+class UserProfileEditView(LoginRequiredMixin, UserPassesTestMixin, UpdateView):
+    template_name = "accounts/profile_edit.html"
+    model = Profile
+    form_class = ProfileForm
+
+    def get_success_url(self):
+        return reverse("accounts:user_profile", kwargs={"pk": self.object.pk})
+
+    def test_func(self):
+        current_user = self.request.user
+        return current_user.pk == self.kwargs["pk"]
```

Comment:
userのpkとprofileのpkが同じとは限らないので，
```suggestion
        return current_user.profile.pk == self.kwargs["pk"]
```
が適切かな